### PR TITLE
add rememberMe for JWT

### LINKED ISSUE #2922: Migrate to ngStorage and add a rememberMe option for JWT authentication
Currently we use angular-local-storage for working with localstorage.

This is the most popular library for dealing with localstorage in AngularJS, but it has a drawback: its API can be configured to use either localstorage or sessionstorage, but not work with both together.

We need this feature if we want to improve JWT authentication:

- Currently the JWT token is stored in localstorage for 24 hours. This isn't good for security, and this isn't good if people want a good "remember me" service.
- My goal is to add a "remember me" box (like we have for session authentication), and store the JWT token in sessionstorage (if remember me is false) or in localstorage (for 30 days, if remember me is true)

In order to use both sessionstorage and localstorage together, I want to migrate from angular-local-storage to [ngStorage](https://github.com/gsklee/ngStorage) which has got the right API for this.

### PR DESCRIPTION
**Pull request by cicoub13**
change angular-local-storage to ngStorage
change authentication-token (name in storage) by ngStorage.authenticationToken
add configuration for timeout in TokenProvider

Fix #2922

### CODE REVIEW

**Comment by cicoub13** [comment]
Do you want i remove angular-local-storage for OAuth2 too ?

**Comment by jdubois** [comment]
Yes please!
Le 18 févr. 2016 2:11 PM, "Cyril Beslay" <notifications@github.com> a
écrit :

**Comment by cicoub13** [comment]
Done.

I don't know why Cucumber fails (but fails in master too).

File: generators/client/templates/src/main/webapp/app/_app.module.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
-angular.module('<%=angularAppName%>', ['LocalStorageModule', <% if (enableTranslation) { %>'tmh.dynamicLocale', 'pascalprecht.translate', <% } %>
-    'ngResource', 'ngCookies', 'ngAria', 'ngCacheBuster', 'ngFileUpload',
+angular.module('<%=angularAppName%>', ['ngStorage', <% if (enableTranslation) { %>'tmh.dynamicLocale', 'pascalprecht.translate', <% } %>
+    'ngResource', 'ngCookies', 'ngAria', 'ngCacheBuster', 'ngFileUpload', <% if (authenticationType === 'jwt') { %>'ngStorage', <% } %>
```
Comment (deepu105) [inline generators/client/templates/src/main/webapp/app/_app.module.js]: why os this added twice?

File: generators/client/templates/src/main/webapp/app/admin/tracker/_tracker.service.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ angular.module('<%=angularAppName%>')
                 var url = '//' + loc.host + loc.pathname + 'websocket/tracker';<% if (authenticationType == 'oauth2') { %>
                 /* globals localStorage */
                 /*jshint camelcase: false */
-                var authToken = JSON.parse(localStorage.getItem('ls.token')).access_token;
+                var authToken = JSON.parse(localStorage.getItem('ngStorage-authenticationToken')).access_token;
```
Comment (deepu105) [inline generators/client/templates/src/main/webapp/app/admin/tracker/_tracker.service.js]: can we use variable name` jhi-authenticationToken` plz

File: generators/client/templates/src/main/webapp/swagger-ui/_index.html [hunk 3]
```diff
@@ -85,18 +85,18 @@
                     window.swaggerUi.api.clientAuthorizations.add("key", apiKeyAuth);
                 <% } %>
                 <% if (authenticationType == 'oauth2') { %>
-                    var authToken = JSON.parse(localStorage.getItem("ls.token")).access_token;
+                    var authToken = JSON.parse(localStorage.getItem("ngStorage-authenticationToken")).access_token;
```
Comment (deepu105) [inline generators/client/templates/src/main/webapp/swagger-ui/_index.html]: `jhi-authenticationToken` plz

**Comment by deepu105** [comment]
I have fixed the cucumber test

Thanks & Regards,
Deepu

**Comment by jdubois** [comment]
Great! I'm merging this as it looks good, I'll test this for real tomorrow

File: generators/client/templates/src/main/webapp/app/admin/tracker/_tracker.service.js [hunk 4]
```diff
@@ -23,7 +23,7 @@ angular.module('<%=angularAppName%>')
                 var url = '//' + loc.host + loc.pathname + 'websocket/tracker';<% if (authenticationType == 'oauth2') { %>
                 /* globals localStorage */
                 /*jshint camelcase: false */
-                var authToken = JSON.parse(localStorage.getItem('ls.token')).access_token;
+                var authToken = JSON.parse(localStorage.getItem('jhi-authenticationToken')).access_token;
```
Comment (deepu105) [inline generators/client/templates/src/main/webapp/app/admin/tracker/_tracker.service.js]: shouldnt this be `$localStorage.authenticationToken` ? @cicoub13

Comment (cicoub13) [inline generators/client/templates/src/main/webapp/app/admin/tracker/_tracker.service.js]: Yes and must include module $localStorage line 5

File: generators/client/templates/src/main/webapp/swagger-ui/_index.html [hunk 5]
```diff
@@ -85,18 +85,18 @@
...
                 <% } %>
                 <% if (authenticationType == 'xauth') { %>
-                    var authToken = JSON.parse(localStorage.getItem("ls.token")).token;
+                    var authToken = JSON.parse(localStorage.getItem("jhi-authenticationToken")).token;
                     var apiKeyAuth = new SwaggerClient.ApiKeyAuthorization("X-Auth-Token", authToken, "header");
                     window.swaggerUi.api.clientAuthorizations.add("key", apiKeyAuth);
                 <% } %>
                 <% if (authenticationType == 'jwt') { %>
-                    var authToken = JSON.parse(localStorage.getItem("ls.authentication-token"));
+                    var authToken = JSON.parse(localStorage.getItem("jhi-authenticationToken"));
```
Comment (cicoub13) [inline generators/client/templates/src/main/webapp/swagger-ui/_index.html]: Bug if user doesn't check "Remember me" at login (token stored in sessionStorage).

**Pull request merged**